tests/e2e/UsingGKE.md

```diff
@@ -0,0 +1,160 @@
+
+* [Step 1: Create a kubernetes cluster](#step-1-create-and-setup-a-kubernetes-cluster)
+* [Step 2: Get cluster credentials](#step-2-get-cluster-credentials)
+* [Step 3: Create Clusterrolebinding](#step-3-create-clusterrolebinding)
+* [Step 4: Export test script variables](#step-4-export-test-script-variables)
+* [Step 5: Run](#step-5-run)
+* [Examples](#examples)
+
+## Step 1: Create and setup a kubernetes cluster
+E2E tests require a Kubernetes cluster. You can create one using the Google Container Engine using
+the following command:
+
+```bash
+gcloud container clusters \
+  create ${CLUSTER_NAME} \
+  --zone ${ZONE} \
+  --project ${PROJECT_NAME} \
+  --cluster-version ${CLUSTER_VERSION} \
+  --machine-type ${MACHINE_TYPE} \
+  --num-nodes ${NUM_NODES} \
+  --enable-kubernetes-alpha \
+  --no-enable-legacy-authorization
+ ```
+
+ - `CLUSTER_NAME`: Whatever suits your fancy, 'istio-e2e' is a good choice.
+ - `ZONE`: 'us-central-f' is a good value to use.
+ - `PROJECT_NAME`: is the name of the GCP project that will house the cluster. You get a project by visiting [GCP](https://console.cloud.google.com).
+ - `CLUSTER_VERSION`: 1.7.3 or later.
+ - `MACHINE_TYPE`: Use 'n1-standard-4'
+ - `NUM_NODES`: Use 3.
+ - `no-enable-legacy-authorization`: Optional, needed if you want to test RBAC.
+
+You must set your default compute service account to include:
+roles/container.admin (Kubernetes Engine Admin)
+Editor (on by default)
+
+To set this, navigate to the IAM section of the Cloud Console and find your default GCE/GKE service account in the
+following form: projectNumber-compute@developer.gserviceaccount.com: by default it should just have the Editor role.
+Then in the Roles drop-down list for that account, find the Kubernetes Engine group and select the role Kubernetes
+Engine Admin. The Roles listing for your account will change to Multiple.
+
+## Step 2: Get cluster credentials
+```
+gcloud container clusters get-credentials ${CLUSTER_NAME} --zone ${ZONE} --project ${PROJECT_NAME}
+```
+
+## Step 3: Create Clusterrolebinding
+```
+kubectl create clusterrolebinding myname-cluster-admin-binding    --clusterrole=cluster-admin    --user="<user_email>"
+```
+* user_email should be the one you use to log in gcloud command. You can do `gcloud info` to find out current user info.
+
+
+## Step 4: Export test script variables
+
+**Option 1:** Build your own images.
+
+```
+# Customize .istiorc.mk (at the top of the istio.io/istio source tree) with your HUB and optional TAG
+# it allows you to customize Makefile rules. For example:
+cat .istiorc.mk
+HUB=costinm
+TAG=mybranch
+
+# Build images on the local docker.
+make docker
+
+# Push images to docker registry
+make push
+
+# the hub/tag set in your .istiorc.mk will be used by the test.
+
+```
+
+**Option 2:** Already committed changes to istio/istio master branch
+NOTE: SHA used as TAG is one that is already committed on istio/istio. You can pick any SHA you want.
+```
+export HUB="gcr.io/istio-testing"
```

Comment:
Not everyone has access to this right? (I know I don't :( )
Should we mention that somewhere here?

Reply:
Maybe add export HUB="docker.io/<account name>" as alternative solution? In this case we need to run docker login first.

Reply:
I think everyone has READ access. These are pushed by CI automatically to GCR.